src/dmd/compiler.d

```diff
@@ -107,38 +111,54 @@ struct Compiler
      */
     extern (C++) static Expression paintAsType(Expression e, Type type)
     {
-        // We support up to 512-bit values.
-        ubyte[64] buffer;
+        union U
+        {
+            d_int32 int32value;
+            d_int64 int64value;
+            float float32value;
+            double float64value;
```

Comment:
This goes against the (near) blanket ban on floating point in the compiler.  But this is code only meant for dmd, so I'll overlook it.

I guess you don't want to support `*(byte[4]*)&fvalue` or `*(float*)&sarray` at compile-time either.

Reply:
LDC has been using this simpler union-based implementatiomn for some time. I didn't get the need for the existing complexity, as there are assertions making sure source/target types are one of [u]int/[u]long/float/double (no static arrays), and that the source/target type sizes match.

Reply:
The door is open for anyone to add support, but no one's taken that baton yet.